src/plan_expand_hypertable.c

```diff
@@ -115,6 +116,93 @@ const_datum_get_int(Const *cnst)
 	pg_unreachable();
 }
 
+/*
+ * Constify expressions of the following form in WHERE clause:
+ *
+ * column OP timestamptz - interval
+ * column OP timestamptz + interval
+ *
+ * Iff interval has no month component.
+ *
+ * Since the operators for timestamptz OP interval are marked
+ * as stable they will not be constified during planning a
+ * subset of these operations are immutable and can be safely
+ * constified during planning. We adjust those expressions here
+ * and transform them for those cases where it's safe to do so
+ * during planning.
+ */
+static OpExpr *
+constify_timestamptz_op_interval(PlannerInfo *root, OpExpr *constraint)
+{
+	Assert(op->args->length == 2);
+
+	Expr *left = linitial(constraint->args);
+	Expr *right = llast(constraint->args);
+	OpExpr *op;
+	bool var_on_left = false;
+	Interval *interval;
+	Const *c_ts, *c_int;
+	Datum constified;
+
+	if (IsA(left, Var) && IsA(right, OpExpr))
+	{
+		op = castNode(OpExpr, right);
+		var_on_left = true;
+	}
+	else if (IsA(left, OpExpr) && IsA(right, Var))
+	{
+		op = castNode(OpExpr, left);
+	}
+	else
+		return constraint;
+
+	Oid plus_oid = ts_get_operator("+", PG_CATALOG_NAMESPACE, TIMESTAMPTZOID, INTERVALOID);
+	Oid minus_oid = ts_get_operator("-", PG_CATALOG_NAMESPACE, TIMESTAMPTZOID, INTERVALOID);
+
+	if (op->opresulttype != TIMESTAMPTZOID || (op->opno != plus_oid && op->opno != minus_oid) ||
```

Comment:
Suggestion: For clarity, I would wrap this condition in an aptly-named function that describes the condition, e.g.,: `is_timestamptz_and_interval_op` or similar. It is a bit hard to follow what this condition is "defining" otherwise.